html/inc/util.inc

```diff
@@ -1010,16 +1010,20 @@ function do_download($path,$name="") {
     readfile($path);
 }
 
-function redirect_to_secure_url($url) {
+// if have SSL but not using it, redirect
+//
+function redirect_to_secure_url() {
     if (defined('SECURE_URL_BASE')
         && strstr(SECURE_URL_BASE, "https://")
-        && (!isset($_SERVER['HTTPS']) || !$_SERVER['HTTPS'])
+        && empty($_SERVER['HTTPS'])
     ) {
-        Header("Location: ".SECURE_URL_BASE."/$url");
+        Header("Location: https://".$_SERVER['HTTP_HOST'].$_SERVER['REQUEST_URI']);
```

Comment:
URL_BASE might have a value of http://www.computeathome.org and SECURE_URL_BASE might have a value of https://secure.computeathome.org.  If a project had done this and their certificate was only good for secure.computeathome.org then this code change would break the site.

Can we use 
Header("Location: ".SECURE_URL_BASE.$_SERVER['REQUEST_URI']);

instead?